synapse/storage/events.py

```diff
@@ -1548,6 +1556,86 @@ def _store_redaction(self, txn, event):
             (event.event_id, event.redacts),
         )
 
+    @defer.inlineCallbacks
+    def _censor_redactions(self):
+        """Censors all redactions older than a month that haven't been censored.
+
+        By censor we mean update the event_json table with the redacted event.
+
+        Returns:
+            Deferred
+        """
+
+        if not self.hs.config.redaction_retention_period:
+            return
+
+        max_pos = yield self.find_first_stream_ordering_after_ts(
+            self._clock.time_msec() - self.hs.config.redaction_retention_period
+        )
+
+        # We fetch all redactions that point to an event that we have that has
+        # a stream ordering from over a month ago, that we haven't yet censored
+        # in the DB.
+        sql = """
+            SELECT er.event_id, redacts FROM redactions
+            INNER JOIN events AS er USING (event_id)
+            INNER JOIN events AS eb ON (er.room_id = eb.room_id AND redacts = eb.event_id)
+            WHERE NOT have_censored
+            AND ? <= er.stream_ordering AND er.stream_ordering <= ?
+            ORDER BY er.stream_ordering ASC
+            LIMIT ?
+        """
+
+        rows = yield self._execute(
+            "_censor_redactions_fetch", None, sql, -max_pos, max_pos, 100
```

Comment:
Maybe add a comment here explaining why we're limiting to `[-max_pose, max_pose]` and why we're only processing the 100 oldest uncensored redactions.